src/AutoMapper/QueryableExtensions/ExpressionBuilder.cs

```diff
@@ -209,15 +209,18 @@ protected override Expression VisitNew(NewExpression node)
         private List<MemberBinding> CreateMemberBindings(ExpressionRequest request, TypeMap typeMap, Expression instanceParameter, TypePairCount typePairCount, LetPropertyMaps letPropertyMaps)
         {
             var bindings = new List<MemberBinding>();
-            foreach (var propertyMap in typeMap.GetPropertyMaps().Where(pm => 
-                (!pm.ExplicitExpansion || request.MembersToExpand.Contains(pm.DestinationProperty)) && pm.CanResolveValue() && ReflectionHelper.CanBeSet(pm.DestinationProperty)))
+            foreach (var propertyMap in typeMap.GetPropertyMaps()
+                                               .Where(pm => (!pm.ExplicitExpansion || request.MembersToExpand.Contains(pm.DestinationProperty)) && 
+                                                            pm.CanResolveValue() && ReflectionHelper.CanBeSet(pm.DestinationProperty))
+                                               .OrderBy(pm => pm.MappingOrder ?? int.MaxValue).ThenBy(pm => pm.DestinationProperty.Name))
```

Comment:
The ordering by MappingOrder is not needed.

Reply:
Removed! However I think that it  would be better that it remained here as mentioned to avoid breaking already implementations using MappingOrder.